### PR TITLE
Create security group to allow ssh from 0.0.0.0

### PR DESCRIPTION
**Pull request by eparis**
This means they will work if the customer does a VPN or VPC pairing. You still
would not be able to access the master or worker by default from the public
internet because they do not have a public IP. It also means that you can create
a machine in the 'public' subnets and access those via ssh.

### CODE REVIEW

**Comment by abhinavdahiya** [comment]
My argument is that we know masters are not going to be moved to public subnets ever. So why open SSH to internet.

**Comment by cuppett** [comment]
It is true it would make it automatically work if the VPC peering or VPN is established for the remote side. However, to do a VPN, they will already be adjusting the other IPI elements (route tables, VPC, etc.). Do we want a rule that they "inherit for free" or by circumstance there or should they be required to overtly add that traffic permission as well? 

The analogy is: Leave the safe unlocked, because if they got into my house, surely they'd be allowed to help themselves to the contents in the safe.

I could go either way, we are halfway there by leaving the rule in there for the VPC CIDR (bastion) instead of specifying the specific security groups only. We should just apply the strategy uniformly once we know and adopt it.

**Comment by eparis** [comment]
ok, steve and I talked offline, I'll come back with another PR.

File: data/data/aws/main.tf [hunk 1]
```diff
@@ -19,11 +19,33 @@ module "bootstrap" {
   target_group_arns        = "${module.vpc.aws_lb_target_group_arns}"
   target_group_arns_length = "${module.vpc.aws_lb_target_group_arns_length}"
   vpc_id                   = "${module.vpc.vpc_id}"
-  vpc_security_group_ids   = "${list(module.vpc.master_sg_id)}"
+  vpc_security_group_ids   = "${list(module.vpc.master_sg_id, aws_security_group.public_ssh.id)}"
```
Comment (wking) [inline data/data/aws/main.tf]: Can you drop [this][1], now that your new group makes it redundant?

[1]: https://github.com/openshift/installer/blob/ee26337a50fc67db480292a07e0a6b0fc8ea17fa/data/data/aws/bootstrap/main.tf#L148-L156

Comment (wking) [reply data/data/aws/main.tf]: In fact, do we want to drop the bootstrap security group altogether and have the bootstrap module add [the journald gateway rule][1] to one of the passed in groups (and have it removed during bootstrap teardown)?  That's more efficient, but somewhat breaks down the isolation between this module and the bootstrap module.

[1]: https://github.com/openshift/installer/blob/ee26337a50fc67db480292a07e0a6b0fc8ea17fa/data/data/aws/bootstrap/main.tf#L158-L166

**Comment by abhinavdahiya** [comment]
> ok, steve and I talked offline, I'll come back with another PR.

Is this still required?

**Comment by abhinavdahiya** [comment]
> > ok, steve and I talked offline, I'll come back with another PR.
> 
> Is this still required?

It looks like this is not required. Feel free to reopen!

/close

**Pull request closed without merging**